File: src/main/java/io/realworld/core/model/Article.java
package io.realworld.core.model;

import java.time.Instant;

public record Article(Long id,
                      String slug,
                      String title,
                      String description,
                      String body,
                      long favoritesCount,
                      Instant createdAt,
                      Instant updatedAt,
                      Profile author) {
}

File: backend/src/main/kotlin/com/hexagonkt/realworld/services/Article.kt
package com.hexagonkt.realworld.services

import java.time.LocalDateTime

data class Article(
    val slug: String,
    val author: String,
    val title: String,
    val description: String,
    val body: String,
    val tagList: Set<String>,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now(),
    val favoritedBy: Set<String> = emptySet(),
    val comments: List<Comment> = emptyList()
)
